pkg/node.go

```diff
@@ -0,0 +1,282 @@
+						"prometheus.io/path":   "/metrics",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Volumes:    createNodeVolumes(),
+					Containers: createNodeContainers(),
+					// todo what is the hack?
+					TerminationGracePeriodSeconds: pointer.Int64Ptr(TerminationGracePeriodSeconds),
+					// todo fill in selectors when passed
+					NodeSelector:       map[string]string{},
+					ServiceAccountName: nodeServiceAccountName,
+					HostIPC:            true,
+				},
+			},
+		},
+	}
+}
+
+func createNodeVolumes() []corev1.Volume {
+	// todo how to avoid this?
+	directory := corev1.HostPathDirectory
+	directoryOrCreate := corev1.HostPathDirectoryOrCreate
+
+	return []corev1.Volume{
+		{Name: LogsVolume, VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		}},
+		{Name: hostDevVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/dev", Type: &directory},
+		}},
+		// todo this if for loopback manager only
+		{Name: hostHomeVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/home", Type: &directory},
+		}},
+		{Name: hostSysVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/sys", Type: &directory},
+		}},
+		{Name: hostRootVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/", Type: &directory},
+		}},
+		{Name: hostRunUdevVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/run/udev", Type: &directory},
+		}},
+		{Name: hostRunLVMVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/run/lvm", Type: &directory},
+		}},
+		{Name: hostRunLock, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/run/lock", Type: &directory},
+		}},
+		{Name: CSISocketDirVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/var/lib/kubelet/plugins/csi-baremetal", Type: &directoryOrCreate},
+		}},
+		{Name: registrationDirVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/var/lib/kubelet/plugins_registry/", Type: &directoryOrCreate},
+		}},
+		{Name: mountPointDirVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/var/lib/kubelet/pods", Type: &directory},
+		}},
+		{Name: csiPathVolume, VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/var/lib/kubelet/plugins/kubernetes.io/csi"},
+		}},
+		// todo optional
+		{Name: driveConfigVolume, VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: loopbackManagerConfigName},
+				Optional:             pointer.BoolPtr(true),
+			},
+		}},
+	}
+}
+
+func createNodeContainers() []corev1.Container {
+	// todo get rid of this hack
```

Comment:
link TODO with the issue?

Reply:
Done. And removed extra TODOs which already planned in issues.